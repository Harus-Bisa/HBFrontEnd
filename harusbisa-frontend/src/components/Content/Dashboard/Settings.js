import React from "react";
import {connect} from "react-redux";
import CourseForm from "../../Form/CourseForm";
import {Input, Button as RButton,FormGroup,Label, Col} from "reactstrap";
import { Button } from "@material-ui/core";

function Settings(props){
    return(
        <div>
            <header style={{borderBottom:'4px solid #F4F4F4'}}>
                <h1>Settings</h1>
            </header>
            <div className="container">
                <div className="row justify-content-center">
                    <div className="col-md-10">
                        <CourseForm header={false} id={props.course.courseId}/>
                        <div className="container-fluid">
                            <div className="content">
                                <FormGroup row>
                                    <Label sm={3}>Kode Kelas</Label>
                                    <Col sm={6}>
                                        <Input value={props.course.joinCode} disabled/>
                                    </Col>
                                    <Col sm={3}>
                                        <RButton className="neutral-button" style={{width:'100%', height:'calc(1.5em + .75rem + 2px)'}}>Salin</RButton>
                                    </Col>
                                </FormGroup>
                                <FormGroup row style={{marginTop:"2rem"}}>
                                    <Col sm={9}>
                                        <h5>Download Daftar Nilai</h5>
                                        <p>Lihat nilai - nilai dan kehadiran siswa anda di kelas ini</p>
                                    </Col>
                                    <Col sm={3}>
                                        <RButton className="neutral-button" style={{width:'100%', height:'calc(1.5em + .75rem + 2px)'}}>Download Excel</RButton>
                                    </Col>
                                </FormGroup>
                                <Button>Download Sesi Kelas</Button>
                            </div>                                
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}

function mapStateToProps(state){
    return{
        course: state.course
    }
}
export default connect(mapStateToProps)(Settings);