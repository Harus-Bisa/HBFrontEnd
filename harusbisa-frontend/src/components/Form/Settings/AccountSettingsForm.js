import React from "react";
import {connect} from "react-redux";
import { Button, Form, FormGroup, Label, Input, Col } from 'reactstrap';

function AccountSettingsForm(props){
    var [firstName, setFirstName] = React.useState(props.firstName);
    var [lastName, setLastName] = React.useState(props.lastName);
    var [school, setSchool] = React.useState(props.school);
    var [email, setEmail] = React.useState(props.email);

    var [oldPassword, setOldPassword] = React.useState();
    var [newPassword, setNewPassword] = React.useState();
    var [verifyNewPassword, setVerifyNewPassword] = React.useState();

    const submitProfile = (event) =>{
        event.preventDefault();
        props.changeProfile()
    }

    if(props.loading){
        return <p>Loading</p>
    }
    return(
        <div>
            <div className="content" style={{borderBottom:"2px solid #F4F4F4", paddingLeft: 0, paddingRight:0}}>
                <div>
                    <h5>Basic</h5>
                </div>
                <div className="row justify-content-end">
                    <div className="col-md-11">
                        <Form onSubmit={submitProfile} id="profile-form">
                            <FormGroup row>
                                <Label sm={5}>Nama Depan</Label>
                                <Col sm={7}>
                                    <Input type="text" id="firstName" value={firstName} onChange={(event) => setFirstName(firstName = event.target.value) }/>
                                </Col>
                            </FormGroup>
                            <FormGroup row>
                                <Label sm={5}>Nama Belakang</Label>
                                <Col sm={7}>
                                    <Input type="text" id="lastName" value={lastName} onChange={(event) => setLastName(lastName = event.target.value)}/>
                                </Col>
                            </FormGroup>
                            <FormGroup row>
                                <Label sm={5}>Nama Universitas</Label>
                                <Col sm={7}>
                                    <Input type="text" id="school" value={school} onChange={(event) => setSchool(school= event.target.value)}/>
                                </Col>
                            </FormGroup>
                            <FormGroup row>
                                <Label sm={5}>Email</Label>
                                <Col sm={7}>
                                    <Input type="email" id="email" value={email} onChange={(event) => setEmail(email = event.target.value)}/>
                                </Col>
                            </FormGroup>
                            <div className="row justify-content-end">
                                <div className="col-sm-7">
                                    <Button type="submit" className="neutral-button" style={{width:'100%'}}>Simpan perubahan</Button>
                                </div>
                            </div>
                        </Form>
                    </div>
                </div>
            </div>
            
            <div className="content" style={{paddingLeft: 0, paddingRight:0}}>
                <h5>Password</h5>
                <div className="row justify-content-end">
                    <div className="col-md-11">
                        <p style={{marginBottom:'2rem'}}>Dengan membuat password memastkan anda dapat log in ke akun Harus Bisa anda</p>
                        <Form onSubmit={submitProfile} id="password-form">
                            <FormGroup row>
                                <Label sm={5}>Password Lama</Label>
                                <Col sm={7}>
                                    <Input type="text" id="oldPassword" value={oldPassword} onChange={(event) => setOldPassword(oldPassword= event.target.value)}/>
                                </Col>
                            </FormGroup>
                            <FormGroup row>
                                <Label sm={5}>Password Baru</Label>
                                <Col sm={7}>
                                    <Input type="text" id="newPassword" value={newPassword} onChange={(event) => setNewPassword(newPassword= event.target.value)}/>
                                </Col>
                            </FormGroup>
                            <FormGroup row>
                                <Label sm={5}>Ulangi Password Baru</Label>
                                <Col sm={7}>
                                    <Input type="text" id="verifyNewPassword" value={verifyNewPassword} onChange={(event) => setVerifyNewPassword(verifyNewPassword= event.target.value)}/>
                                </Col>
                            </FormGroup>
                            <div className="row justify-content-end">
                                <div className="col-sm-7">
                                    <Button type="submit" className="neutral-button" style={{width:'100%'}}>Simpan perubahan</Button>
                                </div>
                            </div>
                        </Form>
                    </div>
                </div>
            </div> 
        </div>
    )
}

function mapStateToProps(state){
    return{
        firstName: state.firstName,
        lastName: state.lastName,
        email: state.email,
        school: state.school,
        loading: state.loading
    }
}

AccountSettingsForm.defaultProps ={
    changeProfile: (() =>{})
}
export default connect(mapStateToProps)(AccountSettingsForm);