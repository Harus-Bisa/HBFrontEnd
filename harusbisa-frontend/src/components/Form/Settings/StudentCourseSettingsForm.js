import React from "react";
import {connect} from "react-redux";
import { Button } from "@material-ui/core";
import StudentCourseForm from "../StudentCourseForm";
import Popup from "../../Popup/Popup";
import CourseSettingsCard from "../../Card/CourseSettingsCard";

function StudentCourseSettingsForm(props){
    const makeCards = ()=>{
        var cards = []
        for (let i=0; i<props.courses.length; i++){
            cards.push(<CourseSettingsCard key={i} id={props.courses[i].courseId}/>)
        }   
        return cards;     
    }
    return(
        <div>
            <div className="content" style={{borderBottom:"2px solid #F4F4F4", paddingLeft: 0, paddingRight:0}}>
                <div>
                    <h5>Kelas</h5>
                </div>
                <div>
                    {makeCards()}
                </div>
            </div>
            <div style={{padding:"2rem", justifyContent:'center', display:'flex'}}>
                <Popup 
                    purpose={"+ Tambah Kelas"} 
                    trigger={{component:Button, className:"student-button"}} 
                    content={StudentCourseForm}
                />
            </div>
        </div>
    )
}

function mapStateToProps(state){
    return{
        courses: state.courses
    }
}

export default connect(mapStateToProps)(StudentCourseSettingsForm);