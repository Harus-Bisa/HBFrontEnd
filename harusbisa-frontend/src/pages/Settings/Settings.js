import React from "react";
import {connect} from "react-redux";
import { withAuth } from "../withAuth";
import AccountSettingsForm from "../../components/Form/Settings/AccountSettingsForm";
import StudentCourseSettingsForm from "../../components/Form/Settings/StudentCourseSettingsForm";
import { Button } from "@material-ui/core";
const ACCOUNT = "ACCOUNT";
const COURSE = "COURSE"

function Settings(props){
    var [settingDisplay, setSettingDisplay] = React.useState(ACCOUNT);
    if (props.loading){
        return null
    }
    const arrowUpStyle = {
        width: 0, 
        height: 0, 
        borderLeft: "5px solid transparent",  
        borderRight: "5px solid transparent",
        borderBottom: "5px solid #f4f4f4",
        fontsize: 0,
        lineheight: 0,
    }
    return(
        <div className="container">
            <div className="row justify-content-center">
                <div className="col-lg-6 col-md-8 col-sm">
                    <div className="content">
                        <div style={{borderBottom:'4px solid #f4f4f4'}}>
                            <div className="row" >
                                <div className="col-md-6">
                                    <h1 style={{margin:'0'}}>Settings</h1>
                                </div>
                                <div className="col-md-6" style={{display:'flex'}}>
                                    <div className="row" style={{margin: "auto"}}>
                                        <div className="col-6">
                                            <Button className= "button settings-button" id={ACCOUNT+"-button"} onClick={() => setSettingDisplay(settingDisplay=ACCOUNT)}>Akun</Button>
                                        </div>
                                        <div className="col-6">
                                            <Button className= "button settings-button" id={COURSE+"-button"} onClick={() => setSettingDisplay(settingDisplay=COURSE)}>Kelas</Button>
                                        </div>
                                    </div>
                                </div>
                            </div>
                            <div className="row justify-content-end">
                                <div className="col-md-6">
                                    <div className="row" style={{margin: "auto"}}>
                                        <div className="col-6">
                                            <div style={{display:'flex', justifyContent:"center"}}><div style={(settingDisplay === ACCOUNT ? arrowUpStyle : {})} id={ACCOUNT}/></div>
                                        </div>
                                        <div className="col-6">
                                            <div style={{display:'flex', justifyContent:"center"}}><div style={(settingDisplay === COURSE ? arrowUpStyle : {})} id={COURSE}/></div>
                                        </div>
                                    </div>
                                </div>
                            </div>
                        </div>

                        <div>
                            {settingDisplay === ACCOUNT && <AccountSettingsForm/>}
                            {settingDisplay === COURSE && <StudentCourseSettingsForm/>}
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}

function mapStateToProps(state){
    return {
        loading: state.loading
    }
}

export default connect(mapStateToProps)(withAuth(Settings));