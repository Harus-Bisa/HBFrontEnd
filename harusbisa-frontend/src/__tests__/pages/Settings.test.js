import React from "react";
import {mount} from "enzyme";
import configureMockStore from "redux-mock-store";
import '../../test-config';
import {Provider} from 'react-redux';
import Settings from "../../pages/Settings/Settings";
import AccountSettingsForm from "../../components/Form/Settings/AccountSettingsForm";
import StudentCourseSettingsForm from "../../components/Form/Settings/StudentCourseSettingsForm";

const mockStore = configureMockStore();
describe("Settings page", () =>{
    let initialState;
    let page;
    let store;
    let history;
    beforeEach(() =>{
        initialState = {
            courses:[{
                "courseId": "5db492c349a67b33b8d0a2a2",
                "courseName": "Calculus I",
                "courseDescription": "The basic of calculus (derivatives).",
                "startTerm": "August 2019",
                "endTerm": "December 2019",
                "numberOfLectures": 5,
                "numberOfStudents": 21,
                "instructors": [
                    "John Doe",
                    "Bryan Smith"
                ],
                "joinCode": "123654"
              }],
              loading: false,
              loggedIn: true,
        }
        history = {
            push: jest.fn()
        }
        localStorage.setItem('id_token', 123);
        store = mockStore(initialState);
        page = mount(<Provider store={store}><Settings history={history}/></Provider>)
    })

    it("Matches snapshot", () =>{
        expect(page.html()).toMatchSnapshot();
    })

    it("renders account form", () =>{
        var accountButton = page.find("#ACCOUNT-button").at(0)
        accountButton.simulate("click")
        page.update()
        var accountForm = page.find(AccountSettingsForm)
        var courseForm = page.find(StudentCourseSettingsForm)
        expect(accountForm).toBeDefined()
        expect(accountForm).toHaveLength(1)
        expect(courseForm).toHaveLength(0) 
    })

    it("renders course form", () =>{
        var courseButton = page.find("#COURSE-button").at(0)
        courseButton.simulate("click")
        page.update()
        var accountForm = page.find(AccountSettingsForm)
        var courseForm = page.find(StudentCourseSettingsForm)
        expect(courseForm).toBeDefined()
        expect(courseForm).toHaveLength(1)
        expect(accountForm).toHaveLength(0) 
    })
})