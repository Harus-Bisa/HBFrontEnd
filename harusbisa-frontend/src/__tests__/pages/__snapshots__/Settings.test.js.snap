// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings page Matches snapshot 1`] = `"<div class=\\"container\\"><div class=\\"row justify-content-center\\"><div class=\\"col-lg-6 col-md-8 col-sm\\"><div class=\\"content\\"><div style=\\"border-bottom: 4px solid #f4f4f4;\\"><div class=\\"row\\"><div class=\\"col-md-6\\"><h1 style=\\"margin: 0px;\\">Settings</h1></div><div class=\\"col-md-6\\" style=\\"display: flex;\\"><div class=\\"row\\" style=\\"margin: auto;\\"><div class=\\"col-6\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text button settings-button\\" tabindex=\\"0\\" type=\\"button\\" id=\\"ACCOUNT-button\\"><span class=\\"MuiButton-label\\">Akun</span><span class=\\"MuiTouchRipple-root\\"></span></button></div><div class=\\"col-6\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text button settings-button\\" tabindex=\\"0\\" type=\\"button\\" id=\\"COURSE-button\\"><span class=\\"MuiButton-label\\">Kelas</span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div></div></div><div class=\\"row justify-content-end\\"><div class=\\"col-md-6\\"><div class=\\"row\\" style=\\"margin: auto;\\"><div class=\\"col-6\\"><div style=\\"display: flex; justify-content: center;\\"><div style=\\"width: 0px; height: 0px; border-left: 5px solid transparent; border-right: 5px solid transparent; border-bottom: 5px solid #f4f4f4;\\" id=\\"ACCOUNT\\"></div></div></div><div class=\\"col-6\\"><div style=\\"display: flex; justify-content: center;\\"><div id=\\"COURSE\\"></div></div></div></div></div></div></div><div><div><div class=\\"content\\" style=\\"border-bottom: 2px solid #F4F4F4; padding-left: 0px; padding-right: 0px;\\"><div><h5>Basic</h5></div><div class=\\"row justify-content-end\\"><div class=\\"col-md-11\\"><form id=\\"profile-form\\" class=\\"\\"><div class=\\"row form-group\\"><label class=\\"col-sm-5 col-form-label\\">Nama Depan</label><div class=\\"col-sm-7\\"><input id=\\"firstName\\" type=\\"text\\" class=\\"form-control\\" value=\\"\\"></div></div><div class=\\"row form-group\\"><label class=\\"col-sm-5 col-form-label\\">Nama Belakang</label><div class=\\"col-sm-7\\"><input id=\\"lastName\\" type=\\"text\\" class=\\"form-control\\" value=\\"\\"></div></div><div class=\\"row form-group\\"><label class=\\"col-sm-5 col-form-label\\">Nama Universitas</label><div class=\\"col-sm-7\\"><input id=\\"school\\" type=\\"text\\" class=\\"form-control\\" value=\\"\\"></div></div><div class=\\"row form-group\\"><label class=\\"col-sm-5 col-form-label\\">Email</label><div class=\\"col-sm-7\\"><input id=\\"email\\" type=\\"email\\" class=\\"form-control\\" value=\\"\\"></div></div><div class=\\"row justify-content-end\\"><div class=\\"col-sm-7\\"><button type=\\"submit\\" style=\\"width: 100%;\\" class=\\"neutral-button btn btn-secondary\\">Simpan perubahan</button></div></div></form></div></div></div><div class=\\"content\\" style=\\"padding-left: 0px; padding-right: 0px;\\"><h5>Password</h5><div class=\\"row justify-content-end\\"><div class=\\"col-md-11\\"><p style=\\"margin-bottom: 2rem;\\">Dengan membuat password memastkan anda dapat log in ke akun Harus Bisa anda</p><form id=\\"password-form\\" class=\\"\\"><div class=\\"row form-group\\"><label class=\\"col-sm-5 col-form-label\\">Password Lama</label><div class=\\"col-sm-7\\"><input id=\\"oldPassword\\" type=\\"text\\" class=\\"form-control\\" value=\\"\\"></div></div><div class=\\"row form-group\\"><label class=\\"col-sm-5 col-form-label\\">Password Baru</label><div class=\\"col-sm-7\\"><input id=\\"newPassword\\" type=\\"text\\" class=\\"form-control\\" value=\\"\\"></div></div><div class=\\"row form-group\\"><label class=\\"col-sm-5 col-form-label\\">Ulangi Password Baru</label><div class=\\"col-sm-7\\"><input id=\\"verifyNewPassword\\" type=\\"text\\" class=\\"form-control\\" value=\\"\\"></div></div><div class=\\"row justify-content-end\\"><div class=\\"col-sm-7\\"><button type=\\"submit\\" style=\\"width: 100%;\\" class=\\"neutral-button btn btn-secondary\\">Simpan perubahan</button></div></div></form></div></div></div></div></div></div></div></div></div>"`;
